warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PushdownUnindexedFieldsVisitor.java

```diff
@@ -0,0 +1,171 @@
+package datawave.query.jexl.visitors;
+
+import static datawave.query.jexl.nodes.QueryPropertyMarker.MarkerType.BOUNDED_RANGE;
+import static datawave.query.jexl.nodes.QueryPropertyMarker.MarkerType.INDEX_HOLE;
+
+import java.util.Set;
+
+import org.apache.accumulo.core.client.TableNotFoundException;
+import org.apache.commons.jexl3.parser.ASTAndNode;
+import org.apache.commons.jexl3.parser.ASTEQNode;
+import org.apache.commons.jexl3.parser.ASTERNode;
+import org.apache.commons.jexl3.parser.ASTReferenceExpression;
+import org.apache.commons.jexl3.parser.JexlNode;
+import org.apache.commons.jexl3.parser.JexlNodes;
+import org.apache.log4j.Logger;
+
+import datawave.query.config.ShardQueryConfiguration;
+import datawave.query.jexl.JexlASTHelper;
+import datawave.query.jexl.LiteralRange;
+import datawave.query.jexl.nodes.QueryPropertyMarker;
+import datawave.query.util.MetadataHelper;
+
+/**
+ * Visitor meant to 'push down' predicates for expressions that are not executable against the index.
+ */
+public class PushdownUnindexedFieldsVisitor extends RebuildingVisitor {
```

Comment:
RebuildingVisitor isn't necessary. If you have 100 terms in a query and only one unindexed term, you're copying 100 terms to swap in a single new term. Extend base visitor and modify the query tree in place, with confidence!

Reply:
I have to use the rebuilding visitor.  Each set of unindexed fields is being applied to the original query plan tree so I cannot modify the tree inline.